src/protocol/context/mod.rs

```diff
@@ -67,3 +67,194 @@ pub trait Context<F: Field>:
     /// Generates a new share of one
     fn share_of_one(&self) -> <Self as Context<F>>::Share;
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::ff::{Field, Fp31};
+    use crate::helpers::Direction;
+    use crate::protocol::malicious::Step::MaliciousProtocol;
+    use crate::protocol::{QueryId, RecordId};
+    use crate::secret_sharing::{MaliciousReplicated, Replicated};
+    use crate::telemetry::metrics::{
+        INDEXED_PRSS_GENERATED, RECORDS_SENT, SEQUENTIAL_PRSS_GENERATED,
+    };
+    use futures_util::future::join_all;
+    use futures_util::try_join;
+    use rand::distributions::{Distribution, Standard};
+    use rand::Rng;
+    use std::iter::repeat;
+
+    use super::*;
+    use crate::test_fixture::{Reconstruct, Runner, TestWorld, TestWorldConfig};
+
+    trait AsReplicated<F: Field> {
+        fn left(&self) -> F;
+        fn right(&self) -> F;
+    }
+
+    impl<F: Field> AsReplicated<F> for Replicated<F> {
+        fn left(&self) -> F {
+            (self as &Replicated<F>).left()
+        }
+
+        fn right(&self) -> F {
+            (self as &Replicated<F>).right()
+        }
+    }
+
+    impl<F: Field> AsReplicated<F> for MaliciousReplicated<F> {
+        fn left(&self) -> F {
+            (self as &MaliciousReplicated<F>).rx().left()
+        }
+
+        fn right(&self) -> F {
+            (self as &MaliciousReplicated<F>).rx().right()
+        }
+    }
+
+    /// Toy protocol to execute PRSS generation and send/receive logic
+    async fn toy_protocol<F, S, C>(ctx: C, index: usize, share: &S) -> Replicated<F>
+    where
+        F: Field,
+        Standard: Distribution<F>,
+        C: Context<F, Share = S>,
+        S: AsReplicated<F>,
+    {
+        let ctx = ctx.narrow("metrics");
```

Comment:
You shouldn't need to narrow at this level.

Reply:
it is not required but I think it is a good practice to narrow when you execute a specific protocol. For example it helps distinguishing access to PRSS from the protocol and malicious context (when it upgrades the input)

Reply:
I'm not sure that I agree, but I won't insist.